### ISSUE TITLE
Add body name to warning and error message

### ISSUE THREAD

**Issue by mgiraud** [godot-jolt/godot-jolt#379]
There are a lot of usefull warning and errors that you have added to your extension.
However when a scene has a lot of physical objects, it's tedious to identify which object causes the warning/error.
If there is not too much performace cost, could you consider adding the name of the object to the message ?

Edit : Printing the hierarchy would be the best, but maybe a bad idea in regards to readability

**Comment by mihe** [comment]
I can only do this for a very limited set of warnings/errors, due to the limited amount of information I have from the physics server. Specifically the ones related to bodies, like warnings about bounce, friction and linear/angular damping. I don't have any sort of instance ID or other reference for things like joints, so I have no way of knowing what object in Godot is using that particular thing on my side.

Hierarchy would technically be possible for those, I think, but I'm not sure what edge-cases there might be in terms of traversing the tree.

Are there any warnings/errors in particular that you run into a lot?

**Comment by mgiraud** [comment]
I've had a lot of 
```
E 0:00:00:0754   build: Failed to build box shape with {half_extents=(0.150000, 0.150000, 0.350000) margin=5.957000}. Its half extents must be greater than its margin.
  <Erreur C++>   Condition "shortest_axis <= margin" is true. Returning: {}
  <Source C++>   shapes\jolt_box_shape_impl_3d.cpp:43 @ build()
```
or similar during my first build because Godot physics does not use this value ^^

But i have this error that i have trouble fixing, i can't find the faulty physics material  
```
W 0:00:00:0745   set_bounce: Bounce values less than 0 or greater than 1 are not supported by Godot Jolt. Values outside this range will be clamped.
  <Source C++>   objects\jolt_body_impl_3d.cpp:821 @ set_bounce()
```

**Comment by mihe** [comment]
I could technically print the names of the bodies that own a particular shape. The problem with that is that you can reuse the same shape across multiple bodies, so you'd potentially have dozens/hundreds/thousands of bodies per message, which would not only be slow to print but also incredibly noisy, much like printing the node hierarchy.

It also wouldn't help with bespoke shapes like the ones you'd create for things like shape-casts and other physics queries.

I'll see about adding the names of the bodies in the messages I mentioned earlier, but that's about as much as I can do I think.

**Comment by mgiraud** [comment]
That would be plenty enough, thanks !

**Comment by mihe** [comment]
It might be worth pointing out that unlike Godot Physics, Jolt allows for a margin (or what it calls "convex radius") of 0 for all applicable shapes, which are boxes, cylinders and convex polygons, although performance might take a slight hit. That would get rid of those shape errors for you.

I should really mention that in the README, I just realized.

**Comment by timshannon** [comment]
FWIW, @mgiraud I ran into the same issue.  The warning also shows up when you open those bodies in the editor, so if you systemically go through and open your scenes you may be able to more easily pin point which bodies have the margin discrepancy.

And like @mihe mentioned, I fixed mine by lowering the margin below my smallest collision shape for those particular bodies.

**Comment by phelioz** [comment]
@mihe Thanks for the tips about margins and thanks for Godot Jolt

In a VR physics project I am working on I have started using Godot Jolt. Like usually in a VR project you have a lot of small objects so getting a lot of errors regarding margin that I didn't get using Godot Physics.

Instead of spamming errors could we not use the maximum allowed margin if the specified margin by the user is too small?
Wouldn't it be a more seamless transition for Godot Physics users in for example VR projects where they wouldn't need to fiddle with the margins for every collision shape.

**Comment by mihe** [comment]
Yeah, you might be right. I'm a bit hesitant about deviating from the value specified in the shape's settings, but I also understand that most people don't even know (or care) what the margin is, let alone what its value is.

Maybe just clamping it is the more pragmatic solution here.

How do you feel about this @jrouwe? Basically what's happening now is that Godot has the convex radius (or what it calls "margin") set to 0.04 for all shapes, but doesn't actually use it for anything, whereas I do, which seems to be creating friction for people migrating over to Jolt because I start throwing warnings and disabling any shape with [an invalid convex radius](https://github.com/jrouwe/JoltPhysics/blob/8d3008e21baff6ce774b821adc73618bb3b891af/Jolt/Physics/Collision/Shape/BoxShape.cpp#L59-L65). So @phelioz is suggesting to basically `margin = min(margin, half_extents)` and just take whatever performance hit comes out of it. Do you see any other negative side effects?

**Comment by mihe** [comment]
Also, unrelated to the previous comment, but I'll take a crack at logging the name of a shape's first owner/body, as well as how many other owners/bodies it belongs to. Hopefully that should make tracking these down a lot easier. I'll see what I can do about query shapes.

**Comment by mihe** [comment]
> in fact the [ConvexHullShape](https://github.com/jrouwe/JoltPhysics/blob/8d3008e21baff6ce774b821adc73618bb3b891af/Jolt/Physics/Collision/Shape/ConvexHullShape.cpp#L269-L344) already has code that does this

Awesome! I was sitting here wondering how I would accommodate for convex hulls. I should have just looked at the code.

> I think you probably meant to subtract epsilon from min_half_extent

Yes, you're absolutely right. I was thinking I'd try to keep the margin as close to the set value as possible. It's probably not a great thing to optimize for though. Your fraction seems like a much better idea.

I guess I can always clarify in the README that the margin is interpreted purely as an upper bound.

**Comment by mihe** [comment]
Warnings about shape margins are gone as of #406.

I can't say exactly when 0.2.0 will drop. I was hoping to get a few more things in before that, but presumably before the end of the week.

**Comment by mihe** [comment]
This fix, along with a bunch of other changes, are included in the just recently published [**0.2.0-rc1**](https://github.com/godot-jolt/godot-jolt/releases/tag/v0.2.0-rc1). I'd appreciate it if you could give it a try. There's a dedicated [discussion](https://github.com/godot-jolt/godot-jolt/discussions/423) open for it if you have questions or concerns unrelated to this issue.
